rainier-core/src/main/scala/com/stripe/rainier/sampler/LeapFrog.scala

```diff
@@ -60,6 +60,85 @@ private[sampler] case class LeapFrog(density: DensityFunction) {
     pqBuf(potentialIndex) = density.density * -1
   }
 
+  /**
+    * Perform l leapfrog steps starting at position q and momentum p
+    * @param l the total number of leapfrog steps to perform
+    * @return the new value of the parameters and momentum
+    * array with updated density
+    */
+  private def steps(l: Int, stepSize: Double): Unit = {
+    initialHalfThenFullStep(stepSize)
+    var i = 1
+    while (i < l) {
+      twoFullSteps(stepSize)
+      i += 1
+    }
+    finalHalfStep(stepSize)
+  }
+
+  private def isUTurn(theta: Array[Double], pqP: Array[Double]): Boolean = {
```

Comment:
Given that the other methods in the call chain just directly use `pqBuf`, is there any reason to pass it in as an arg here?